src/TodoList.tsx

```diff
@@ -0,0 +1,54 @@
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { useLocation } from 'react-router-dom';
+import { Todo } from './react-app-env';
+import { TodoItem } from './TodoItem';
+
+type Props = {
+  todos: Todo[];
+  setTodos: (todos: Todo[]) => void;
+};
+
+export const TodoList: React.FC<Props> = ({ todos, setTodos }) => {
+  const [editTodoId, setEditTodoId] = useState(0);
+  const { pathname } = useLocation();
+  let todosFilter = todos;
+
```

Comment:
Consider wrapping the filter function content into a useCallback or useEffect to avoid unnecessary re-renders 🤔